### PR TITLE
Enforcing arches version compatibility in a project, re #8702

### PR DESCRIPTION
**Pull request by chiatt**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

- Allows an admin to define a project version and enforce min and max arches version compatibility 
- Displays project version with Arches version under the DEBUG flag in main navigation (lower left)
- Ensures new projects automatically define as the min and max allowable version of arches, the version of arches used to create the project.

### Issues Solved
#8702

### CODE REVIEW

File: arches/app/utils/compatibility.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    """
+
+    if min_arches is None or max_arches is None:
+        logger.error(
+            _("Either a minimum or maximum compatible Arches version is not specified. Unable to check {0} compatibility".format(target))
+        )
+        return
+
+    if len(min_arches) <= 3 and len(max_arches) <= 3:
```
Comment (apeters) [inline arches/app/utils/compatibility.py:26]: I'm not sure that we should **require** that a project creator specify both a min and max compatible version, or either for that matter.  A required min version might make sense, but a max version is harder to understand.  
At the moment packages aren't pip installed, so upgrading to keep packages and arches versions in sync is more difficult.  

Was any thought put to using a library like [this](https://github.com/python-semver/python-semver) to manage version numbers and comparisons?

File: arches/app/templates/base-manager.htm [hunk 2]
```diff
@@ -272,7 +272,10 @@ <h4 style="flex: 1" class="ep-form-alert-title" data-bind="text: alert().title">
                 {% if app_settings.DEBUG %}
                 <div class="debug-notice">
                     <div>{% trans 'DEBUG' %}</div>
+                    <div>{% trans 'Arches' %}</div>
```
[the remark this answers is not in the document]
Comment (chiatt) [reply arches/app/templates/base-manager.htm]: Good point - I've fixed it.

File: arches/app/utils/compatibility.py (hunk 1 above)
Comment (apeters) [reply arches/app/utils/compatibility.py:26]: @chiatt Yeah, that all makes sense to me.

**Comment by chiatt** [comment]
To QA:

run `pip install -r requirements.txt`

Next add this setting to your project:
APP_VERSION = semantic_version.Version(major=1, minor=0, patch=0, prerelease=('a', '0'))
Confirm it displays like this at the bottom of the nav bar:
image

Next add these settings to your project. Note that the beta version is lower than 
MIN_ARCHES_VERSION = "7.4.0-b0"
MAX_ARCHES_VERSION = "7.4.0"

Not that the app still runs as expected (on the dev/7.4.x branch)
Change the MAX_ARCHES_VERSION to 7.3.0 and Arches should stop running. You should get an error message like this: ```arches.app.utils.compatibility.CompatibilityError: 'This project is incompatible with Arches 7.4.0b0.'```

**Review by aarongundel — APPROVED** [review]
Works as described for me.  My only complaint is that it feels a little heavy handed to have the application fail to start.  Ideally it would give something more visual for feedback rather than failing without anything being served up.  That being said, this is an advanced feature, at least it's logging it, and I don't expect many people will know about it - shouldn't affect more than a handful of projects once it gets out into the wild.  Therefore, 💯

**Comment by chiatt** [comment]
Thanks @aarongundel - I totally agree about having Arches fail to start. I agree a more visual approach would be better. Maybe we can add that refinement in a patch release.

**Pull request merged**